### ISSUE TITLE
libpals: provide proper `cpus_per_pe`

### ISSUE THREAD

**Issue by jameshcorbett** [flux-framework/flux-coral2#29]
According to David Gloe at HPE, the [cpus_per_pe member](https://github.com/flux-framework/flux-coral2/blob/63ba22ec96de394588731cf3d15b2d90157f8941/src/shell/plugins/cray_pals.c#L109) of the `pals_cmd_t` struct should be "the number of Linux CPUs (hyperthreads) each PE [i.e. shell task] is bound to." He also noted that

> Overlapping isn't taken into account at all for this. So for example if there are 4 PEs all bound to the same 2 hyperthreads, cpus_per_pe should be set to 2.

The value the flux-coral2 shell plugin provides is not correct because it uses the notion of hwloc cores, not Linux CPUs, and it _does_ account for overlap.

@grondo noted that 

> The distribution of cores to tasks is done in the shell affinity plugin, and as such it is not exported to other plugins at the moment... if you attempt to recreate it, you'd have to use the same code

Thankfully it seems that an improper value (e.g. 0) for the `cpus_per_pe` entry does not cause any errors, and David Gloe guessed that the value isn't read anywhere. So we can punt on the issue for a little while.

**Comment by grondo** [comment]
Just to clarify, by "shell tasks" you mean the user tasks that the job shell launches, not the job shell itself correct?

**Comment by jameshcorbett** [comment]
Yeah, sorry. I was looking for the word and didn't want to say "MPI rank". I should have just said "task" since that's the standard Flux terminology.
